src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import java.util.ArrayList;
+
+class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Creates a new task list.
+     */
+    TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Creates a new task list using an existing ArrayList.
+     *
+     * @param tasks ArrayList of tasks.
+     */
+    TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Delete a task from the task list.
+     *
+     * @param taskId Task ID.
+     * @return Task that was deleted.
+     * @throws DukeException If ID is not found.
+     */
+    Task deleteTask(int taskId) throws DukeException {
+        try {
+            return this.tasks.remove(taskId - 1);
+        } catch (Exception e) {
+            throw new DukeException("Failed to find task.");
+        }
+    }
+
+    /**
+     * Mark a task as done.
+     *
+     * @param taskId Task ID.
+     * @return Task that was marked done.
+     * @throws DukeException If ID is not found.
+     */
+    Task markDone(int taskId) throws DukeException {
+        try {
+            Task task = this.tasks.get(taskId - 1);
+            task.markAsDone();
+            return task;
+        } catch (Exception e) {
+            throw new DukeException("Failed to find task.");
+        }
+    }
+
+    /**
+     * Add a task to the task list.
+     *
+     * @param task Task to be added.
+     * @return Flag whether task was successfully added.
+     */
+    boolean addTask(Task task) {
+        return this.tasks.add(task);
+    }
+
+    /**
+     * Access underlying ArrayList for export purposes.
+     *
+     * @return ArrayList of tasks.
+     */
+    ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Generates a numbered list of tasks matching a query.
```

Comment:
This description does not seem to tally with what the method is doing

Reply:
Hmm, could you elaborate on this please?

Reply:
Ah, I was mistaken. The return description gave me the impression that it would return a List object instead of the String representation of a list.